gtsam/hybrid/tests/testHybridNonlinearFactorGraph.cpp

```diff
@@ -0,0 +1,761 @@
+
+  EXPECT_LONGS_EQUAL(ghfg.size(), 1);
+}
+
+/****************************************************************************
+ * Test construction of switching-like hybrid factor graph.
+ */
+TEST(HybridFactorGraph, Switching) {
+  Switching self(3);
+
+  EXPECT_LONGS_EQUAL(7, self.nonlinearFactorGraph.size());
+  EXPECT_LONGS_EQUAL(7, self.linearizedFactorGraph.size());
+}
+
+/****************************************************************************
+ * Test linearization on a switching-like hybrid factor graph.
+ */
+TEST(HybridFactorGraph, Linearization) {
+  Switching self(3);
+
+  // Linearize here:
+  HybridGaussianFactorGraph actualLinearized =
+      self.nonlinearFactorGraph.linearize(self.linearizationPoint);
+
+  EXPECT_LONGS_EQUAL(7, actualLinearized.size());
+}
+
+/****************************************************************************
+ * Test elimination tree construction
+ */
+TEST(HybridFactorGraph, EliminationTree) {
+  Switching self(3);
+
+  // Create ordering.
+  Ordering ordering;
+  for (size_t k = 1; k <= self.K; k++) ordering += X(k);
+
+  // Create elimination tree.
+  HybridEliminationTree etree(self.linearizedFactorGraph, ordering);
+  EXPECT_LONGS_EQUAL(1, etree.roots().size())
+}
+
+/****************************************************************************
+ *Test elimination function by eliminating x1 in *-x1-*-x2 graph.
+ */
+TEST(GaussianElimination, Eliminate_x1) {
+  Switching self(3);
+
+  // Gather factors on x1, has a simple Gaussian and a mixture factor.
+  HybridGaussianFactorGraph factors;
+  // Add gaussian prior
+  factors.push_back(self.linearizedFactorGraph[0]);
+  // Add first hybrid factor
+  factors.push_back(self.linearizedFactorGraph[1]);
+
+  // TODO(Varun) remove this block since sum is no longer exposed.
+  // // Check that sum works:
+  // auto sum = factors.sum();
+  // Assignment<Key> mode;
+  // mode[M(1)] = 1;
+  // auto actual = sum(mode);               // Selects one of 2 modes.
+  // EXPECT_LONGS_EQUAL(2, actual.size());  // Prior and motion model.
+
+  // Eliminate x1
+  Ordering ordering;
+  ordering += X(1);
+
+  auto result = EliminateHybrid(factors, ordering);
+  CHECK(result.first);
+  EXPECT_LONGS_EQUAL(1, result.first->nrFrontals());
+  CHECK(result.second);
+  // Has two keys, x2 and m1
+  EXPECT_LONGS_EQUAL(2, result.second->size());
+}
+
+/****************************************************************************
+ * Test elimination function by eliminating x2 in x1-*-x2-*-x3 chain.
+ *                                                m1/      \m2
+ */
+TEST(HybridsGaussianElimination, Eliminate_x2) {
+  Switching self(3);
+
+  // Gather factors on x2, will be two mixture factors (with x1 and x3, resp.).
+  HybridGaussianFactorGraph factors;
+  factors.push_back(self.linearizedFactorGraph[1]);  // involves m1
+  factors.push_back(self.linearizedFactorGraph[2]);  // involves m2
+
+  // TODO(Varun) remove this block since sum is no longer exposed.
```

Comment:
remove

Reply:
Yeah I kept this since I wanted to highlight this change in the PR